pkg/controllers/cassandra/actions/create_nodepool.go

```diff
@@ -0,0 +1,38 @@
+package actions
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1"
+	"github.com/jetstack/navigator/pkg/controllers"
+	"github.com/jetstack/navigator/pkg/controllers/cassandra/nodepool"
+)
+
+type CreateNodePool struct {
+	Cluster  *v1alpha1.CassandraCluster
+	NodePool *v1alpha1.CassandraClusterNodePool
+}
+
+var _ controllers.Action = &CreateNodePool{}
+
+func (a *CreateNodePool) Name() string {
+	return "CreateNodePool"
+}
+
+func (a *CreateNodePool) Execute(s *controllers.State) error {
+	ss := nodepool.StatefulSetForCluster(a.Cluster, a.NodePool)
+	_, err := s.Clientset.AppsV1beta1().StatefulSets(ss.Namespace).Create(ss)
+	if k8sErrors.IsAlreadyExists(err) {
+		return nil
```

Comment:
There are a few good reasons for them to be idempotent:
 1. the `CassandraCluster.Status.NodePools` map (upon which this action is calculated) may not have been updated.
 2. future actions (e.g. ScaleIn) may perform a more complex sequence of steps which need to be "ratcheted" through, by repeated calls to `ScaleIn.Execute`, e.g.
   1. Decommission node
   2. Decrease the statefulset replica count.
  In which case each of those intermediate steps must be idempotent and in which case the final step may as well be idempotent.
 3. The lister used to calculate the action may not have the most up-to-date  resources, so it appears there isn't a StatefulSet, until you attempt to create it, which fails with AlreadyExists (although I might be making that up, because the current implementation doesn't check the lister first)

On the whole, it makes sense to me that these actions should all be idempotent, but happy to discuss further.